### ISSUE TITLE
Training Nan after some epochs

### ISSUE THREAD

**Issue by zhangpeng2001** [ultralytics/ultralytics#14950]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When i train the yolo-v8-m-obb in my custom dataset, i face the Training Nan problem. The loss is regular in the beginning epochs. However, the loss will be nan after some epochs. The epoch that losses are nan will be larger when i lower the learning rate. Why does this happen?

**Comment by Kayzwer** [comment]
it is gradient exploding, try to lower learning rate, use a smaller model, or use small imgsz

**Comment by zhangpeng2001** [comment]
i wonder to know why the smaller model or smaller imgsz will remit the problem? What could be the reason？

**Comment by Y-T-G** [comment]
What's the command you're using for training?

**Comment by zhangpeng2001** [comment]
This is my training command.
import argparse
from ultralytics import YOLO

def parse_args():
    parser = argparse.ArgumentParser(description='Train a detector')
    parser.add_argument('config', help='config-yaml')
    parser.add_argument('data', help='data-yaml')
    parser.add_argument('pretrain', help='pretrian model')
    parser.add_argument('--epochs', default=100, help='training epochs')
    parser.add_argument('--imgsz', default=1024, help='image size for training the model')
    parser.add_argument('--batch', default=32, help='default batch size')
    parser.add_argument('--lr0', default=0.0016, help='default learning rate')
    parser.add_argument('--works', default=8, help='default works per GPU')
    parser.add_argument('--save-period', default=10, help='default preiod to save model')
    parser.add_argument('--eval-epoch', default=10, help='default preiod to evaluation')
    args = parser.parse_args()
    return args

args = parse_args()
config_yaml = args.config
data_yaml = args.data
pretrian_model = args.pretrain

# Load a model
model = YOLO(config_yaml)   # build a new model from YAML
model = YOLO(pretrian_model) # load a pretrained model
model = YOLO(config_yaml).load(pretrian_model)  # build from YAML and transfer weights

model_name = config_yaml.split('/')[-1][:-5]

# Train the model
results = model.train(data=data_yaml, epochs=args.epochs, imgsz=args.imgsz, batch=args.batch, workers=args.works,
                      optimizer='AdamW', lr0=0.00001*args.batch, cos_lr=True, save_period=args.save_period, val_period=args.eval_epoch, 
                      name=model_name, device=[0,1])

**Comment by Y-T-G** [comment]
Try using SGD
